### DISCUSSION TITLE
How to use a custom column for roles in an existing production schema?

### DISCUSSION THREAD

**Discussion by Prinmode** [spatie/laravel-permission#2718] [Q&A]
I'm currently integrating the spatie/laravel-permission package into an existing production system that already has a roles column in the users table. This column is being used to manage basic roles in the application, such as "supervisor" or "executive", through custom logic.

Since the package documentation specifies that there shouldn't be a pre-existing roles column, I'm facing a conflict. Changing the column name across the system is complicated due to the current production environment, so I'd like to know if there's a way to configure the package to use a custom column name instead of the default roles.

Is there any way to modify the package configuration or extend the models to use a different column name for roles? If not, do you have any recommended approaches for handling this kind of situation in an existing schema?

Thank you in advance!

**Comment by parallels999** [comment]
>I'd like to know if there's a way to configure the package to use a custom column name instead of the default roles

roles is not a column, it's an another independent table, but it is using `roles` as laravel relation name, you could change `roles` table name, but not the laravel relation name `roles`
